src/full/Agda/Syntax/Translation/AbstractToConcrete.hs

```diff
@@ -29,6 +30,10 @@ import Control.Applicative hiding (empty)
 import Control.Monad.Reader
 import Control.Monad.State
 
+#if __GLASGOW_HASKELL__ >= 800
+import qualified Control.Monad.Fail as Fail
+#endif
```

Comment:
This `#ifdef` and the one in the instance declaration below can become unconditional now that #3728 has been merged in. And `LANGUAGE CPP` can stay out above.